### ISSUE TITLE
exception while reading jornal file encoded in utf-8

### ISSUE THREAD

**Issue by truekabal** [EDCD/EDMarketConnector#779]
**Please complete the following information:**
 - Version 4.1.4
 - OS: Windows 10
 - OS Locale: Russian
 - Firefox Browser
 - [EDMarketConnector-debug.log](https://github.com/EDCD/EDMarketConnector/files/5625260/EDMarketConnector-debug.log)
 - [journals.zip](https://github.com/EDCD/EDMarketConnector/files/5625303/journals.zip)
 - Game language is russian
   
**Describe the bug**
I have configured EDMC application. it's authorized at frontier services and synced with inara e.t.c
Windows app EDMarketConnector.exe works well.

I tried to use Trade Computer Extension MK.II with EDMC and caught an exception during EDMC.exe launch
It sounds like 
```
2020-12-01 23:02:56.362 - ERROR - EDMC.main:195: Can't read Journal file
Traceback (most recent call last):
  File "EDMC.py", line 188, in main
  File "encodings\cp1251.pyc", line 23, in decode
UnicodeDecodeError: 'charmap' codec can't decode byte 0x98 in position 4300: character maps to <undefined>
```

It also fails if i launch EDMC.exe from command promt as well

**To Reproduce**
1. play Elite dangerous with russian language (or try to use attached journals)
2. run EDMC.exe
3. See error

**Comment by A-UNDERSCORE-D** [comment]
Please provide a full log. This looks to be weirdness with windows encodings which may be changed depending on locales.

**Comment by truekabal** [comment]
> 
> 
> Please provide a full log. This looks to be weirdness with windows encodings which may be changed depending on locales.

edmc log file is attached if you meant it. if not - In which file can i get full log?

**Comment by A-UNDERSCORE-D** [comment]
I see, sorry, I was apparently tired when looking into it. Im reading logs now

**Comment by A-UNDERSCORE-D** [comment]
Yeah we've had this issue before. Can you update your windows to the latest version? We already handle this, but older versions of windows 10 have issues with locales.

**Comment by truekabal** [comment]
Did you mean [this issue](https://github.com/EDCD/EDMarketConnector/issues/711)?
But as i said, windowed application works well, 

The problem appears if i use python 'open' method on files encoded in utf-8
I reproduced it it even in a raw python console.

Are you sure, that windows updates will solve this issue?

Can you please try opening journal log files i attached in your python env with 'open' method without setting encoding param?

**Comment by Athanasius** [comment]
You skipped/deleted this part of the bug report template:

Always attach `%TEMP%\EDMarketConnector.log` from *immediately* after the bug occurs (re-running the application overwrites this file).

Yes, we really do need both log files.  This one includes *all* output, which as well as raw print()'s also means un-caught exception traces.  The -debug log file only includes output that went through the logging system.

The core issue here is you're one of the systems (a VERY old Windows 10 version, although in this case it's still in support) that doesn't support setting the Unicode Code Page (65001) *and* then fails to set a UTF-8 encoding in our code.

Personally I'm not inclined to add another "but what if we're not UTF-8?" hack to the code as would be required here.  What we'd probably need to do is record the locale encoding after trying (whether we succeed or not) to set UTF-8 encoding, and reference that *everywhere* we open a file.  Unless, that is, testing shows that in the case where we *have* set UTF-8 we can do without explicitly specifying it, then those arguments to things like `open()` calls could be removed.

The real, clean, solution that we can be sure won't cause any further issues is for everyone to be on at least Windows 10 1903 which is when the UTF-8 code page was added.

**Comment by truekabal** [comment]
Okay, i got the point.
I'll attach %TEMP%\EDMarketConnector.log after reproducing an issue and will check/update windows.

thanks.

**Comment by Athanasius** [comment]
Except now I just read more closely and see this is an issue with EDMC.py, that means both those logfiles are irrelevant.   Instead we need whatever full output you see on a CMD.EXE when running the same command as "Trade Computer Extension MK.II " attempts.

It being in EDMC.py, we're not even trying to use UTF-8 encoding on the relevant `open()` call.  It might perhaps help if we do.

**Comment by truekabal** [comment]
> both those logfiles are irrelevant
Raw EDMC.exe run in CMD shows only described exception, nothing else. 

Also it adds this info to attached log file. So I can't say it's irrelevant actually. It contains log of raw EDMC.exe run via command promt

**Comment by Athanasius** [comment]
This should now be working properly with the just-released 4.1.5
